src/js/launcher.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env node */
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var kettle = fluid.registerNamespace("kettle");
+
+fluid.require("%kettle/lib/KettleConfigLoader.js");
+
+var yargs   = require("yargs");
+var path    = require("path");
+var process = require("process");
+
+fluid.registerNamespace("gpii.launcher");
+
+/**
+ *
+ * Merge options from (in order of precedence):
+ *
+ * 1. An options file, if supplied.
+ * 2. Environment variables.
+ * 3. Command line parameters.
+ *
+ * @param that - The `gpii.launcher` component itself.
+ *
+ */
+gpii.launcher.launchComponent = function (that) {
```

Comment:
This is part-way to what I imagined. Rather than manually stuffing things into the configuration from this loader, they would be available as IoC-resolved material within the configuration itself, and the launching process would just consist of an invocation of the standard Kettle command-line driver as seen in https://github.com/fluid-project/kettle/blob/master/init.js or https://github.com/GPII/universal/blob/master/index.js#L50 . Compare with https://github.com/GPII/first-discovery-server/blob/master/src/js/configUtils.js#L21-L32 for a baseline